cmd/cored/main.go

```diff
@@ -73,6 +74,17 @@ var (
 
 	race          []interface{} // initialized in race.go
 	httpsRedirect = true        // initialized in plain_http.go
+
+	// By default, the mock HSM is enabled for use.
+	// This feature is turned off for production binaries
+	// with the no_mockshm build tag.
+	enableMockHSM = func(db pg.DB) []core.RunOption {
+		return []core.RunOption{core.MockHSM(mockhsm.New(db))}
```

Comment:
This is a great idea but I think we should still try to avoid linking the mockhsm package in this mode, so we probably need to back to two files, `mockhsm.go` and `no_mockhsm.go`. :/

And in that case, they can go back to ordinary function definitions.